scripts/update_boot_iso

```diff
@@ -0,0 +1,265 @@
+#!/usr/bin/python3
+
+import argparse
+import os
+import shutil
+import time
+import sys
+import subprocess
+
+# Absolute path to the main project directory
+PROJECT_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+
+# Relative path to the ISO folder within the project
+ISO_FOLDER = os.path.join(PROJECT_DIR, "result", "iso")
+
+# Initial boot ISO we will update
+INPUT_ISO = os.path.join(ISO_FOLDER, "boot.iso")
+INPUT_ISO_REVISION_FILE = os.path.join(ISO_FOLDER, "boot.iso.git_rev")
+
+# Updated boot ISO (including Anaconda updates image and possibly other bits)
+UPDATED_ISO = os.path.join(ISO_FOLDER, "updated_boot.iso")
+
+# Folder needed to include updates image into the updated ISO
+UPDATES_FOLDER = os.path.join(ISO_FOLDER, "images")
+UPDATES_IMAGE = "updates.img"
+BOOT_OPTIONS = "inst.sshd inst.nokill"
+
+# Command names
+MKKSISO = "mkksiso"
+VIRT_INSTALL = "virt-install"
+
+def warmup_sudo():
+    """Get sudo right after invocation, for use later (mkksiso needs it).
+    otherwise we might get into the situation where sudo is requested in the
+    middle of execution of this script (e.g., after a long makeupdates run if widget compilation
+    is necessary), destroying the UX.
+    """
+    os.system('sudo echo "we need sudo later, so lets get it now"')
+
+def get_first_non_upstream_commit():
+    """Get the first commit that is not upstream."""
+    try:
+        result = subprocess.run(['git', 'rev-list', '--no-merges', '--first-parent', 'HEAD'], capture_output=True, text=True, check=True)
+        commits = result.stdout.strip().split('\n')
+        if commits:
+            return commits[-1]
+    except subprocess.CalledProcessError:
+        print("** error: could not determine the first non-upstream commit")
+        sys.exit(1)
+
+    return None
+
+def make_updates_image(git_id):
+    """Build an updates image based on tag/hash and prepare it for inclusion in boot ISO.
+
+    :param str git_id: git revision id (hash, tag, etc.)
+    """
+    if git_id is None:
+        print("** make updates:git_id is None, falling back to finding first non-upstream commit id")
+        git_id = get_first_non_upstream_commit()
+        if git_id is None:
+            print("** error: could not determine a valid commit id")
+            sys.exit(1)
+
+    print("** preparing updates image via tag/hash: %s" % git_id)
```

Comment:
Also, don't want to be picky, but looking at the code convention guideline, it states that we should use `format()` instead of `%s`. 

Since this is a script that is mainly intended for development purposes, is it fine if we just ignore that convention?

https://github.com/rhinstaller/anaconda/blob/master/CONTRIBUTING.rst#code-conventions

Reply:
Oh, I saw that in line 120 you are using f-strings. So maybe this is some older code.